datafusion/sql/tests/cases/plan_to_sql.rs

```diff
@@ -240,6 +240,35 @@ fn roundtrip_statement_with_dialect() -> Result<()> {
             parser_dialect: Box::new(GenericDialect {}),
             unparser_dialect: Box::new(UnparserDefaultDialect {}),
         },
+        // more tests around subquery/derived table roundtrip
+        TestStatementWithDialect {
+            sql: "SELECT string_count FROM (
+                    SELECT
```

Comment:
I recommend you add a query that has an additional test that selects more than one column and an expression

something like

```sql
select string_count * 100, id FROM ...
```

Reply:
Thanks for the suggestion. This query is actually not working as "CAST" will be presented in the inner projection and the projections matching actually breaks.

Maybe another PR to fix this case.

Reply:
It might also be worth filing an issue to track the defect